lisa/lisa-core/src/main/java/it/unive/lisa/analysis/string/CharInclusion.java

```diff
@@ -0,0 +1,297 @@
+package it.unive.lisa.analysis.string;
+
+import it.unive.lisa.analysis.Lattice;
+import it.unive.lisa.analysis.SemanticDomain;
+import it.unive.lisa.analysis.SemanticDomain.Satisfiability;
+import it.unive.lisa.analysis.SemanticException;
+import it.unive.lisa.analysis.nonrelational.value.BaseNonRelationalValueDomain;
+import it.unive.lisa.analysis.representation.DomainRepresentation;
+import it.unive.lisa.analysis.representation.StringRepresentation;
+import it.unive.lisa.program.cfg.ProgramPoint;
+import it.unive.lisa.symbolic.value.Constant;
+import it.unive.lisa.symbolic.value.operator.binary.BinaryOperator;
+import it.unive.lisa.symbolic.value.operator.binary.StringConcat;
+import it.unive.lisa.symbolic.value.operator.binary.StringContains;
+import it.unive.lisa.symbolic.value.operator.binary.StringEndsWith;
+import it.unive.lisa.symbolic.value.operator.binary.StringEquals;
+import it.unive.lisa.symbolic.value.operator.binary.StringIndexOf;
+import it.unive.lisa.symbolic.value.operator.binary.StringStartsWith;
+import it.unive.lisa.symbolic.value.operator.ternary.TernaryOperator;
+import it.unive.lisa.symbolic.value.operator.unary.UnaryOperator;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * The suffix string abstract domain.
+ *
+ * @author <a href="mailto:vincenzo.arceri@unipr.it">Vincenzo Arceri</a>
+ * @author <a href="mailto:sergiosalvatore.evola@studenti.unipr.it">Sergio
+ *             Salvatore Evola</a>
+ * 
+ * @see <a href=
+ *          "https://link.springer.com/chapter/10.1007/978-3-642-24559-6_34">
+ *          https://link.springer.com/chapter/10.1007/978-3-642-24559-6_34</a>
+ */
+public class CharInclusion extends BaseNonRelationalValueDomain<CharInclusion> {
+
+	private final Collection<Character> certainlyContained;
+
+	private final Collection<Character> maybeContained;
+
+	private static final CharInclusion TOP = new CharInclusion();
+	private static final CharInclusion BOTTOM = new CharInclusion(null, null);
+
+	/**
+	 * Builds the top char inclusion abstract element.
+	 */
+	public CharInclusion() {
+		this(new HashSet<>(), getAlphabet());
+	}
+
+	/**
+	 * Builds a char inclusion abstract element.
+	 *
+	 * @param certainlyContained the set of certainly contained characters
+	 * @param maybeContained     the set of maybe contained characters
+	 */
+	public CharInclusion(Collection<Character> certainlyContained,
+			Collection<Character> maybeContained) {
+		this.certainlyContained = certainlyContained;
+		this.maybeContained = maybeContained;
+	}
+
+	@Override
+	public CharInclusion lubAux(CharInclusion other) throws SemanticException {
+		Collection<Character> lubAuxCertainly = new HashSet<>();
+
+		Collection<Character> lubAuxMaybe = new HashSet<>(this.getMaybeContained());
```

Comment:
Valid here and in the other classes: as those getters just return the field, it's faster to access the field directly instead of calling the method

Reply:
@lucaneg Thank you for the review. Can I remove the getters methods then?

Reply:
No leave them. We might want to accesss the sets from checks or other domains!